### ISSUE TITLE
User deactivation does not deactivates user sessions

### ISSUE THREAD

**Issue by LeonidChistov** [ory/kratos#1811]
**Describe the bug**

When user is deactivated via administrative API, all existing user sessions remain active (in terms of DB column flag). That looks illogical and is potentially error prone.

**Reproducing the bug**

1. Create new user and login under the new user account
2. Deactivate new user
3. Check `sessions` table: user session will remain active

**Expected behavior**

All sessions of deactivated user must be moved to inactive state.

**Comment by aeneasr** [comment]
Yes, the user "active" state is just a dummy for now, setting it has no effect!

**Comment by aeneasr** [comment]
Hm I think this has a couple of points where we would need to have access control in place. Maybe we could solve it at the identity persister level? The problem is that the identity is used almost everywhere so ensuring it can’t do anything is tricky.

One easy solution could be to disallow it in the session manager

**Comment by frederikhors** [comment]
Please do not stale it.

**Comment by aeneasr** [comment]
This actually already works. An inactive user is not able to check the sessions. The SQL table remains untouched, but we use the identity's state value as part of the session check.

**Comment by aeneasr** [comment]
Currently, the expand parameter is always included though, so this is indeed fixed.
